cmd/atlas/internal/lint/lint.go

```diff
@@ -269,34 +269,23 @@ func (d *DevLoader) inspect(ctx context.Context) (*schema.Realm, error) {
 	return d.Dev.InspectRealm(ctx, opts)
 }
 
-func stmts(f migrate.File) ([]*migrate.Stmt, error) {
-	if s, ok := f.(interface {
-		StmtDecls() ([]*migrate.Stmt, error)
-	}); ok {
-		return s.StmtDecls()
-	}
-	s1, err := f.Stmts()
-	if err != nil {
-		return nil, err
+// lock database so no one else interferes with our change detection.
+func (d *DevLoader) lock(ctx context.Context) (schema.UnlockFunc, error) {
+	l, ok := d.Dev.Driver.(schema.Locker)
+	if !ok {
+		return nil, errors.New("driver does not support locking")
 	}
-	s2 := make([]*migrate.Stmt, len(s1))
-	for i := range s1 {
-		p, err := pos(f, s1[i])
-		if err != nil {
-			return nil, err
-		}
-		s2[i] = &migrate.Stmt{Pos: p, Text: s1[i]}
+	name := "atlas_lint"
```

Comment:
this prevents other linters from running, but not migrations from executing, right?

Reply:
right